src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.storage;
+
+import java.util.Arrays;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Entity;
+
+/**
+ * Contains the information for an individual step of a build.
+ */
+@Entity(name = "buildStep")
+public class BuildStep {
+
+  // Order of the step
+  private final int stepNumber;
+
+  // BuildStep name (e.g "clean-src-dir")
+  private final String name;
+
+  // BuildStep text (e.g "clean-src-dir skipped")
+  private final String text;
+
+  // Indicator of whether or not the step is complete
+  private final boolean isFinished;
+
+  // Indicator of whether or not the step has started
+  private final boolean isStarted;
+
+  // All the logs associated with this step
+  private final <List>Log logs;
+
+  /**
+   * Constructs an instance of BuildStep. name, text and logs can be null.
+   */
+  BuildStep(String name, String[] text, boolean isFinished, boolean isStarted, Log[] logs) {
+    this.name = name;
+    this.text = Arrays.toString(text);
+    this.isFinished = isFinished;
+    this.isStarted = isStarted;
+    this.logs = Arrays.asList(logs);
+  }
+
+  /**
+   * Returns the name of the log as a String. Can be null.
+   */
+  public String getName() {
```

Comment:
Consider adding `@NonNull` [annotation](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/lang/NonNull.html#:~:text=Annotation%20Type%20NonNull&text=A%20common%20Spring%20annotation%20to,infer%20nullability%20of%20Spring%20API.) for the return types.

Reply:
I was under the assumption name could be null here!

Reply:
I don't have a counter example so i'll add the annotation.

Reply:
Annoted with commit 6d47eb4